Revert "gigamento changes reserve.sol"

bayological opened the pull request:
### Description

The changes in [f7b79d0](https://github.com/celo-org/celo-monorepo/commit/f7b79d0a15ba1e783fa30db948b94c4b932c92a4) were mistakenly added to the main branch and do not reflect the protocol as it is currently deployed. This change removes the additional code that was added to the reserve smart contract and tests.

### Other changes

- N/A

### Tested

- Ran existing test suite to verify functionality works.

### Related issues

- Closes #168 

### Backwards compatibility

- N/A

### Documentation

- N/A

bowd reviewed (changes requested):
Looks good overall. One minor change tho, you should also revert the version number of this contract to what's on celo-monorepo: `(1, 1, 2, 2)`. We changed it to `(1,2,0,0)` at some point.

bayological commented:
> Looks good overall. One minor change tho, you should also revert the version number of this contract to what's on celo-monorepo: `(1, 1, 2, 2)`. We changed it to `(1,2,0,0)` at some point.

Thanks, reverted this too.
